### DISCUSSION TITLE
ParserBuilder.ZipOOOOOO only can handle 6 Parser,But i have too many header fileds(more than 6)

### DISCUSSION THREAD

**Discussion by No-Karma** [pointfreeco/swift-url-routing#46] [Q&A]
code:
```
    Route(.case(MTServerRoute.api)) {
                Path{ "api" }
                Parse(.memberwise(MTServerRoute.Api.init(a:b:c:d:e:f:route:))){
                    Headers  {
                        Field("a-1",default: "")
                        Field("a-2",default: "")
                        Field("a-3",default: "")
                        Field("a-4",default: "")
                        Field("a-5",default: "")
                        Field("a-6",default:false) { Bool.parser()}
                    }
                     self.apiRouter
                }
            }
```
error:
`Cannot convert value of type '(String, String, String, String, String, Bool, MTServerRoute.Api.Route) -> MTServerRoute.Api' to expected argument type '(((String, String, String, String, String, Bool), MTServerRoute.Api.Route)) -> MTServerRoute.Api'`

**Comment by oliverfoggin** [comment]
I don't think that is the issue you're facing here.

The `Headers` parsing is returning a tuple of `(String, String, String, String, String, Bool)`. And then you have the `self.apiRouter` returning a `Route`.

So this code is trying to init the `MTServerRoute.Api` with two parameters... `headers` (the tuple) and `route` (the Route).

But... your init method takes seven parameters (each of the six headers and then the route).

So, you have a mismatch in the types. I'm not sure if there is a way to "un-group" the headers but you might be best updating the init to take like...

```
init(headers: (String, String, String, String, String, Bool), route: Route) {}
```

Possibly wrapping the `headers` into a typealias or a struct or something.
